fpmsyncd/routesync.cpp

```diff
@@ -144,6 +207,226 @@ void RouteSync::parseEncap(struct rtattr *tb, uint32_t &encap_value, string &rma
+                                        string &node_len, string &func_len,
+                                        string &arg_len)
+{
+    struct rtattr *tb_my_sid_format[256] = {};
+    uint8_t block_len_buf, node_len_buf, func_len_buf, arg_len_buf;
+
+    parseRtAttrNested(tb_my_sid_format, 4, tb);
+
+    if (tb_my_sid_format[SRV6_LOCALSID_FORMAT_BLOCK_LEN])
+    {
+        block_len_buf = *(uint8_t *)RTA_DATA(
+            tb_my_sid_format[SRV6_LOCALSID_FORMAT_BLOCK_LEN]);
+        block_len += to_string(block_len_buf);
+    }
+    else
+    {
+        block_len += DEFAULT_SRV6_MY_SID_BLOCK_LEN;
+    }
+
+    if (tb_my_sid_format[SRV6_LOCALSID_FORMAT_NODE_LEN])
+    {
+        node_len_buf = *(uint8_t *)RTA_DATA(
+            tb_my_sid_format[SRV6_LOCALSID_FORMAT_NODE_LEN]);
+        node_len += to_string(node_len_buf);
+    }
+    else
+    {
+        node_len += DEFAULT_SRV6_MY_SID_NODE_LEN;
+    }
+
+    if (tb_my_sid_format[SRV6_LOCALSID_FORMAT_FUNC_LEN])
+    {
+        func_len_buf = *(uint8_t *)RTA_DATA(
+            tb_my_sid_format[SRV6_LOCALSID_FORMAT_FUNC_LEN]);
+        func_len += to_string(func_len_buf);
+    }
+    else
+    {
+        func_len += DEFAULT_SRV6_MY_SID_FUNC_LEN;
+    }
+
+    if (tb_my_sid_format[SRV6_LOCALSID_FORMAT_ARG_LEN])
+    {
+        arg_len_buf = *(uint8_t *)RTA_DATA(
+            tb_my_sid_format[SRV6_LOCALSID_FORMAT_ARG_LEN]);
+        arg_len += to_string(arg_len_buf);
+    }
+    else
+    {
+        /* arg_len is optional, by default arg_len is 0 */
+        arg_len += DEFAULT_SRV6_MY_SID_ARG_LEN;
+    }
+
+    SWSS_LOG_INFO("Rx Srv6 MySid block_len:%s node_len:%s func_len:%s arg_len:%s",
+                  block_len.c_str(), node_len.c_str(), func_len.c_str(),
+                  arg_len.c_str());
+
+    return true;
+}
+
+/**
+ * @parseSrv6MySid() - Parses sRv6 MySid attributes
+ * @tb:         Pointer to rtattr to look for nested items in.
+ * @block_len:  (output) locator block length
+ * @node_len:   (output) locator node length
+ * @func_len:   (output) function length
+ * @arg_len:    (output) argument length
+ * @action:     (output) behavior defined for the MySID.
+ * @vrf:        (output) VRF name.
+ * @adj:        (output) adjacency.
+ *
+ * Return:      true on success, false otherwise.
+ */
+bool RouteSync::parseSrv6MySid(struct rtattr *tb[], string &block_len,
+                                  string &node_len, string &func_len,
+                                  string &arg_len, string &action,
+                                  string &vrf, string &adj)
+{
+    uint32_t action_buf = SRV6_LOCALSID_ACTION_UNSPEC;
+    char vrf_buf[IFNAMSIZ + 1] = {0};
+    char adj_buf[MAX_ADDR_SIZE + 1] = {0};
+
+    if (tb[SRV6_LOCALSID_FORMAT])
+    {
+        if (!parseSrv6MySidFormat(tb[SRV6_LOCALSID_FORMAT], block_len,
+                                node_len, func_len, arg_len))
+        {
+            SWSS_LOG_ERROR("Invalid Srv6 MySid format");
```

Comment:
could you add details about node/block/function in the error log message

Reply:
Added. Thanks for the comment.